### PR TITLE
Docker containers for PHP 7.1 development

### PR DESCRIPTION
**Pull request by TomHAnderson**
Rather than force a contributing user to wire their bare-metal, use these containers for easy development.

### CODE REVIEW

File: DOCKER.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+Development and testing using Docker
```
Comment (Ocramius) [inline DOCKER.md]: This should not land in this module unless we:

 1. Really need it (we don't: this is an integration layer between a framework and a library)
 2. We also use it to run our builds (also here: travis-ci already provides the infrastructure, and while not really stable, it's what the ODM maintainers are used to)

Comment (TomHAnderson) [reply DOCKER.md]: 1. A mongodb server is required to run unit tests
2. If I install mongodb on the php container we could closely simulate travis-ci; we could certainly more-closely simulate it than the current non-option.

p.s. I did the same thing here: https://github.com/zfcampus/zf-apigility-doctrine/pull/308 and while I don't expect concurrent agreement between these projects the vote, currently, is one 👍 one 👎

Comment (Ocramius) [reply DOCKER.md]: How to set up a dev environment should really not be provided here: it is addressed ops work and maintenance that will produce more work down the line. Let's just keep it out of the responsibility of this package. Linking the official mongodb image is more than sufficient.

Comment (TomHAnderson) [reply DOCKER.md]: 👍 er, I mean 👎 no, wait, that's a "roger that"

**Pull request closed without merging**